extensions/integration-tests/src/schemaCompare.test.ts

```diff
@@ -251,11 +255,73 @@ class SchemaCompareTester {
+		assert(excludeResult.success === false, 'Exclude should have been unsuccessful');
+		assert(excludeResult.blockingDependencies.length === 1, `There should be one blocking dependency. Actual: ${excludeResult.blockingDependencies.length}`);
+		assert(excludeResult.blockingDependencies[0].sourceValue[1] === 'v1', `Blocking dependency should be view v1. Actual ${excludeResult.blockingDependencies[0].sourceValue[1]}`);
+
+		// Exclude the view v1 that t2 was a dependency for and it should succeed and t2 should also be excluded
+		const v1Difference = schemaCompareResult.differences.find(e => e.sourceValue && e.sourceValue[1] === 'v1' && e.name === 'SqlView');
+		assert(v1Difference !== undefined, 'The difference View v1 should be found. Should not be undefined');
+		const excludeResult2 = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, v1Difference, false, azdata.TaskExecutionMode.execute);
+		assert(excludeResult2.success === true, 'Exclude should have been successful');
+		assert(excludeResult2.affectedDependencies.length === 1, `There should be one affected dependency. Actual: ${excludeResult2.affectedDependencies.length}`);
+		assert(excludeResult2.affectedDependencies[0].sourceValue[1] === 't2', `Table t2 should be the affected dependency. Actual: ${excludeResult2.affectedDependencies[0].sourceValue[1]}`);
+		assert(excludeResult2.affectedDependencies[0].included === false, 'Table t2 should be excluded as a result of excluding v1. Actual: true');
+
+		// including the view v1 should also include the table t2
+		const includeResult = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, v1Difference, true, azdata.TaskExecutionMode.execute);
+		assert(includeResult.success === true, 'Include should have been successful');
+		assert(includeResult.affectedDependencies.length === 1, `There should be one affected dependency. Actual: ${includeResult.affectedDependencies.length}`);
+		assert(includeResult.affectedDependencies[0].sourceValue[1] === 't2', `Table t2 should be the affected dependency. Actual: ${includeResult.affectedDependencies[0].sourceValue[1]}`);
+		assert(includeResult.affectedDependencies[0].included === true, 'Table t2 should be include as a result of including v1. Actual: false');
+
+		// excluding views from the comparison should make it so t2 can be excluded
+		deploymentOptions.excludeObjectTypes.push(mssql.SchemaObjectType.Views);
+		await schemaCompareService.schemaCompare(operationId, source, target, azdata.TaskExecutionMode.execute, deploymentOptions);
+		const excludeResult3 = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, t2Difference, false, azdata.TaskExecutionMode.execute);
+		assert(excludeResult3.success === true, 'Exclude should have been successful');
+		assert(excludeResult3.affectedDependencies.length === 0, `There should be no affected dependencies. Actual: ${excludeResult3.affectedDependencies}`);
```

Comment:
good coverage! 
nit : can we combine the four checks into a function with different parameters?

Reply:
I combined a few of the common ones but left the more specific checks where they were